src/sortStudents.ts

```diff
@@ -1,16 +1,71 @@
-
-export interface Student {
-  // describe Student interface
+interface Student {
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade
 }
 
-// create SortOrder type
-export type SortOrder;
+type SortOrder = 'asc'|'desc';
+
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
+  let result: Student[];
 
+  switch (sortBy) {
+    case SortType.Name:
+      result = (order === 'asc')
+        ? [...students].sort((student1, student2) => (
+          student1.name.localeCompare(student2.name)))
+        : [...students].sort((student1, student2) => (
+          student2.name.localeCompare(student1.name)));
+      break;
+    case SortType.Surname:
+      result = (order === 'asc')
+        ? [...students].sort((student1, student2) => (
+          student1.surname.localeCompare(student2.surname)))
+        : [...students].sort((student1, student2) => (
+          student2.surname.localeCompare(student1.surname)));
+      break;
```

Comment:
```suggestion
    case SortType.Name:
    case SortType.Surname:
      result = (order === 'asc')
        ? [...students].sort((student1, student2) => (
          student1.name.localeCompare(student2.name)))
        : [...students].sort((student1, student2) => (
          student2.name.localeCompare(student1.name)));
      break;
```
Код дублюється. Давай об'єднаємо